File: src/Guias/TP4/Ej9/Accounts.java
package Guias.TP4.Ej9;

public abstract class Accounts {
    private final int id;
    private double balance;
    Accounts(int id){
        this.id=id;
        this.balance=balance;
    }
    public void deposit(double amount){
        this.balance=+amount;
    }
    public void extract (double amount){
        if (canExtract(amount)){
            this.balance -= amount;
        }
        else{
            throw new RuntimeException("Extract cannot be done, no enough founds");
        }
    }
    public double getBalance() {
        return balance;
    }
    public int getId() {
        return id;
    }
    protected abstract boolean canExtract(double amount);

    @Override
    public String toString() {
        return "Id:%d Balance:%.2f".formatted(id,balance);
    }
}

File: Ruby/Guias/G7/Ej5/Accounts.rb
class Accounts
  def initialize
    raise NotImplementedError.new("#{self.class.name} is an abstract class.")
  end
  def deposit(amount)
    @balance +=amount
  end
  def withdraw(amount)
    @balance -=amount
  end
  def get_balance
    @balance
  end
  def to_s
    "Cuenta #{@id} con saldo #{@balance}"
  end
  protected
  def withdraw?(amount)
    raise NotImplementedError.new("#{self.class.name} is an abstract class.")
  end
end